### ISSUE TITLE
iRODS account creation fails on www with 1.16 deploy

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#2969]
As part of post deploy checks the following error occurs in attempting to create an iRODS account for a user account just created.
![image](https://user-images.githubusercontent.com/3456833/47749241-579aec80-dc52-11e8-9c93-426335dc1285.png)

**Comment by sblack-usu** [comment]
I'm not seeing anything that sticks out in the logs... these entries look suspicious, but I'm not familiar with iRODS account creation.
```
[30/Oct/2018 16:22:32] ERROR [django.security.DisallowedHost:204] Invalid HTTP_HOST header: '152.54.2.89'. You may need to add u'152.54.2.89' to ALLOWED_HOSTS.
[30/Oct/2018 16:22:32] ERROR [django.request:256] Internal Server Error: /
[30/Oct/2018 16:39:37] ERROR [django.security.DisallowedHost:204] Invalid HTTP_HOST header: '152.54.2.89'. You may need to add u'152.54.2.89' to ALLOWED_HOSTS.
[30/Oct/2018 16:39:37] ERROR [django.request:256] Internal Server Error: /
[30/Oct/2018 16:39:38] ERROR [django.security.DisallowedHost:204] Invalid HTTP_HOST header: '152.54.2.89'. You may need to add u'152.54.2.89' to ALLOWED_HOSTS.
[30/Oct/2018 16:39:38] ERROR [django.request:256] Internal Server Error: /alexa
[30/Oct/2018 16:39:39] ERROR [django.security.DisallowedHost:204] Invalid HTTP_HOST header: '152.54.2.89'. You may need to add u'152.54.2.89' to ALLOWED_HOSTS.
[30/Oct/2018 16:39:39] ERROR [django.request:256] Internal Server Error: /
[30/Oct/2018 17:07:07] ERROR [django.security.DisallowedHost:204] Invalid HTTP_HOST header: 'hydroshare.cuahsi.org'. You may need to add u'hydroshare.cuahsi.org' to ALLOWED_HOSTS.
```

**Comment by dtarb** [comment]
This issue may relate to #2816 and #2906.  @hyi can you investigate

**Comment by hyi** [comment]
@dtarb I will investigate further whenever I have time today. My preliminary investigation showed the account dtcc123 account has been indeed created successfully in iRODS user zone, so I have to investigate the error on Django side.

**Comment by hyi** [comment]
@dtarb I tested iRODS user creation on beta and everything just works, but I can reproduce the irods user creation issue on production. So it is not a code-related issue, but a configuration issue in our production environment, probably iRODS-related. When I run create_user script manually on the user zone server, I got the error CAT_INVALID_USER when calling ichmod for modifying access control, and this all works on test servers. I will have to investigate further on that iRODS error trying to get to the bottom of that.

**Comment by hyi** [comment]
@dtarb I have just fixed this issue. For some reason, the rods service account in data zone disappeared in user zone, which is referred to by create irods user script to make this rods service account in data zone own the new irods user account home directory. After I recreated this rods service account of data zone in the user zone, it worked immediately. I have tested using a new account on www and before my fix, I ran into the same error as you, and after the fix, it created the corresponding iRODS user account successfully. Please test it out using your dtcc123 or a new account to verify the fix.

**Comment by dtarb** [comment]
I can now successfully create an iRODS account.  But when I delete it I briefly get the error message below, then my profile switches out of edit mode.  The account appears to be deleted, as I can no longer log in to iRODS.  But the error message creates worry that something has not worked right.
![image](https://user-images.githubusercontent.com/3456833/47866259-db281b00-ddc3-11e8-8132-07c61376511b.png)

Note that the checklist item #2934 said create irods account, test connection and delete irods account.  The delete part is not working smoothly.

**Comment by dtarb** [comment]
@hyi could you or @phuongdm or @cbcunc take steps to make sure the rods service account you mention does not disappear in the future, especially with the iRODS update being planned.

**Comment by hyi** [comment]
@dtarb I tried to delete my test iRODS account and it shows success message, so I cannot reproduce the error you saw. Can you try to recreate this irods account and try to delete it again to see if you can reproduce this error consistently? I don't seem to be able to reproduce the delete irods account issue. I remembered it used to stay in edit mode after creating or deleting an irods account. Perhaps the recent front end changes modified the behavior to make it go to the view mode after any editing, but this would be a separate issue if we want to make it stay in edit mode.

**Comment by dtarb** [comment]
@hyi I am able to repeat it.  I had to repeat it a few times to get the screen shot because the message is only on the screen for a few seconds.  However to double check, I just created a new account in HydroShare with a username I have not used before (dddtest), created and deleted an iRODS account and I still get the error message.  Going on a hunch, I then tried from "Chrome".  On the Chrome browser, I do not get this error, but I do reproducibly get it in Firefox.
